[RAG] Agentic RAG workflow on tabular data from a PDF file

AgentGenie opened the pull request:
<!-- Thank you for your contribution! Please review https://docs.ag2.ai/docs/contributor-guide/contributing before opening a pull request. -->

<!-- Please add a reviewer to the assignee section when you create a PR. If you don't have the access to it, we will shortly find a reviewer and assign them to your PR. -->

## Why are these changes needed?
Introduce a workflow to extract accurate tabular data information from a pdf file.
This is one key component to support agentic rag on general pdf file.

<!-- Please give a short summary of the change and the problem this solves. -->

## Related issue number
Related to #68
<!-- For example: "Closes #1234" -->

## Checks

- [x] I've included any doc changes needed for https://docs.ag2.ai/. See https://docs.ag2.ai/docs/contributor-guide/documentation to build and test documentation locally.
- [ ] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [x] I've made sure all auto checks have passed.

marklysze commented:
Thanks @AgentGenie, nice work on getting a workflow to successfully extract data from a table image within a PDF!

A couple of notes:
- can you please add the requirement to install with the neo4j extra: `pip install ag2[neo4j]`
- For the last bullet point in the top section can you please indicate (in better words than mine :) ) that the agentic workflow uses a RAG agent is used to extract document metadata (image of table data based on table name), table image to markdown through a multi-modal agent, and an LLM to answer the question.
- For the `Parse PDF file` section, I know it has a note on price but you please update to include an estimated cost (e.g. you noted $15USD) to run. Similarly, it mentions to skip unless you insist, but can you tell them where to skip to (I think the cell starting with `# IMPORTS`)

Thanks!

AgentGenie commented:
@marklysze Addressed the comments. Pre-commit from my local looks fine. I guess it is because git lfs but don't know how to resolve it. Can we skip it?
<img width="855" alt="Screenshot 2025-01-07 at 11 28 02 PM" src="https://github.com/user-attachments/assets/347b489e-352c-4fe1-9abf-c67c39acf826" />

marklysze commented:
> @marklysze Addressed the comments. Pre-commit from my local looks fine. I guess it is because git lfs but don't know how to resolve it. Can we skip it? <img alt="Screenshot 2025-01-07 at 11 28 02 PM" width="855" src="https://private-user-images.githubusercontent.com/189036189/401048395-347b489e-352c-4fe1-9abf-c67c39acf826.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzYzNjQxOTIsIm5iZiI6MTczNjM2Mzg5MiwicGF0aCI6Ii8xODkwMzYxODkvNDAxMDQ4Mzk1LTM0N2I0ODllLTM1MmMtNGZlMS05YWJmLWM2N2MzOWFjZjgyNi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMTA4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDEwOFQxOTE4MTJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zYTc0NzhhYWY1ODFjYzJhZGI4YTA0ZjQzNTNlNzk5N2YzMDE2MmZkYmY0ZjJlMDBjNWI5NGYzMDExNmIyMjNjJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.6dbxdMNu7Us1bO-qq_gA_hxDmoeTuZzw-VVXEl3JA2k">

I'll look at ignoring these in the pre-commit config.

marklysze reviewed (approved):
Nice work @AgentGenie!
